lib/remote-ftp.js

```diff
@@ -400,9 +404,13 @@ module.exports = {
 
 		var self = this;
 
-		//if (!atom.project.remoteftp.isConnected())
-		//	return;
-
+		if (!atom.project.remoteftp.isConnected()) {
+			if(atom.config.get('Remote-FTP.autoConnectOnSave'))
```

Comment:
First of all I don't think it's good style to encapsulate another `if` here. I would just use:
```
if (!atom.project.remoteftp.isConnected() && !atom.config.get('Remote-FTP.autoConnectOnSave'))
    return;
```
You don't need to connect here, `atom.project.remoteftp.upload(...)` does this for you.
You can just do a code cleanup commit on this or I just merge the few lines manually :)